thetis/callback.py

```diff
@@ -89,20 +90,27 @@ def __init__(self, filename, varnames, array_dim=1, attrs=None,
                     hdf5file.create_dataset('time', (0, 1),
                                             maxshape=(None, 1), dtype=dtype)
                 for var in self.varnames:
-                    hdf5file.create_dataset(var, (0, array_dim),
-                                            maxshape=(None, array_dim), dtype=dtype)
+                    dim_list = array_dim
+                    if isinstance(dim_list, tuple):
+                        dim_list = list(array_dim)
+                    elif not isinstance(dim_list, list):
+                        dim_list = list([array_dim])
+                    shape = tuple([0] + dim_list)
+                    max_shape = tuple([None] + dim_list)
```

Comment:
Why is this inside the loop - none of it depends on `var` - unless I'm missing something...